python/paddle/fluid/contrib/tests/test_calibration.py

```diff
@@ -0,0 +1,229 @@
+    width, height = img.size
+    size = target_size
+    if center == True:
+        w_start = (width - size) / 2
+        h_start = (height - size) / 2
+    else:
+        w_start = np.random.randint(0, width - size + 1)
+        h_start = np.random.randint(0, height - size + 1)
+    w_end = w_start + size
+    h_end = h_start + size
+    img = img.crop((w_start, h_start, w_end, h_end))
+    return img
+
+
+def process_image(sample, mode, color_jitter, rotate):
+    img_path = sample[0]
+
+    img = Image.open(img_path)
+
+    img = resize_short(img, target_size=256)
+    img = crop_image(img, target_size=DATA_DIM, center=True)
+
+    if img.mode != 'RGB':
+        img = img.convert('RGB')
+
+    img = np.array(img).astype('float32').transpose((2, 0, 1)) / 255
+    img -= img_mean
+    img /= img_std
+
+    return img, sample[1]
+
+
+def _reader_creator(file_list,
+                    mode,
+                    shuffle=False,
+                    color_jitter=False,
+                    rotate=False,
+                    data_dir=DATA_DIR):
+    def reader():
+        with open(file_list) as flist:
+            full_lines = [line.strip() for line in flist]
+            if shuffle:
+                np.random.shuffle(full_lines)
+
+            lines = full_lines
+
+            for line in lines:
+                img_path, label = line.split()
+                #img_path = img_path.replace("JPEG", "jpeg")
+                img_path = os.path.join(data_dir, img_path)
+                if not os.path.exists(img_path):
+                    continue
+                yield img_path, int(label)
+
+    mapper = functools.partial(
+        process_image, mode=mode, color_jitter=color_jitter, rotate=rotate)
+
+    return paddle.reader.xmap_readers(mapper, reader, THREAD, BUF_SIZE)
+
+
+def val(data_dir=DATA_DIR):
+    file_list = os.path.join(data_dir, 'val_list.txt')
+    return _reader_creator(file_list, 'val', shuffle=False, data_dir=data_dir)
+
+
+class TestCalibration(unittest.TestCase):
+    def setUp(self):
+        # Download and unzip test data set
+        imagenet_dl_url = 'http://paddle-inference-dist.bj.bcebos.com/int8/calibration_test_data.tar.gz'
+        zip_file_name = imagenet_dl_url.split('/')[-1]
+        cmd = 'rm -rf data {}  && mkdir data && wget {} && tar xvf {} -C data'.format(
+            zip_file_name, imagenet_dl_url, zip_file_name)
+        os.system(cmd)
+        # resnet50 fp32 data
+        resnet50_fp32_model_url = 'http://paddle-inference-dist.bj.bcebos.com/int8/resnet50_int8_model.tar.gz'
+        resnet50_zip_name = resnet50_fp32_model_url.split('/')[-1]
+        resnet50_unzip_folder_name = 'resnet50_fp32'
+        cmd = 'rm -rf {} {} && mkdir {} && wget {} && tar xvf {} -C {}'.format(
+            resnet50_unzip_folder_name, resnet50_zip_name,
+            resnet50_unzip_folder_name, resnet50_fp32_model_url,
+            resnet50_zip_name, resnet50_unzip_folder_name)
+        os.system(cmd)
```

Comment:
It seems this UT runs 5 minutes+, and it costs the second largest elapsed time.
```
[08:26:32]	[Step 1/1] 498/535 Test #535: test_calibration ................................   Passed  336.08 sec
```

Please put the download process in cmake, and put them in `"${THIRD_PARTY_PATH}/int8"` to cache it. Thus, CI system only needs download once, to decrease the download and unzip time.

http://ci.paddlepaddle.org/viewLog.html?buildId=51697&tab=buildLog&buildTypeId=Paddle_PrCi&logTab=tree&filter=all

You can refer to 
https://github.com/PaddlePaddle/Paddle/blob/f534c66d2d7d87aa580538513f4835439acd7bc0/paddle/fluid/inference/tests/test.cmake#L1-L16

Reply:
> It seems this UT runs 5 minutes+, and it costs the second largest elapsed time.
> 
> ```
> [08:26:32]	[Step 1/1] 498/535 Test #535: test_calibration ................................   Passed  336.08 sec
> ```
> Please put the download process in cmake, and put them in `"${THIRD_PARTY_PATH}/int8"` to cache it. Thus, CI system only needs download once, to decrease the download and unzip time.
> 
> http://ci.paddlepaddle.org/viewLog.html?buildId=51697&tab=buildLog&buildTypeId=Paddle_PrCi&logTab=tree&filter=all
> 
> You can refer to
> [Paddle/paddle/fluid/inference/tests/test.cmake](https://github.com/PaddlePaddle/Paddle/blob/f534c66d2d7d87aa580538513f4835439acd7bc0/paddle/fluid/inference/tests/test.cmake#L1-L16)
> 
> Lines 1 to 16 in [f534c66](/PaddlePaddle/Paddle/commit/f534c66d2d7d87aa580538513f4835439acd7bc0)
> 
>  set(INFERENCE_URL "http://paddle-inference-dist.cdn.bcebos.com" CACHE STRING "inference download url") 
>  set(INFERENCE_DEMO_INSTALL_DIR "${THIRD_PARTY_PATH}/inference_demo" CACHE STRING 
>      "A path setting inference demo download directories.") 
>  function (inference_download install_dir url filename) 
>      message(STATUS "Download inference test stuff from ${url}/${filename}") 
>      file(DOWNLOAD "${url}/${filename}" "${install_dir}/${filename}") 
>      message(STATUS "finish downloading ${filename}") 
>  endfunction() 
>   
>  function (inference_download_and_uncompress install_dir url filename) 
>      inference_download(${install_dir} ${url} ${filename}) 
>      execute_process( 
>              COMMAND ${CMAKE_COMMAND} -E tar xzf ${install_dir}/${filename} 
>              WORKING_DIRECTORY ${install_dir} 
>      ) 
>  endfunction()

Add the TODO label in the code and will update the code in another PR.